rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp

```diff
@@ -110,10 +115,42 @@ struct avc2_settings
 			ar(voice_muting_players, voice_muting, video_muting, speaker_muting, speaker_volume_level);
 		}
 	}
-};
 
-// Callback handle tag type
-struct avc2_cb_handle_t{};
+	void register_cb_call(u32 event, u32 error_code)
+	{
+		// This is equivalent to the dispatcher code
+		sysutil_register_cb_with_id<avc2_cb_handle_t>([=, this](ppu_thread& cb_ppu) -> s32
+			{
+				vm::ptr<CellSysutilAvc2Callback> avc2_cb{};
+				vm::ptr<void> avc2_cb_arg{};
+
+				{
+					std::lock_guard lock(this->mutex_cb);
+					avc2_cb = this->avc2_cb;
+					avc2_cb_arg = this->avc2_cb_arg;
+				}
+
+				if (avc2_cb)
+				{
+					avc2_cb(cb_ppu, event, error_code, avc2_cb_arg);
+
+					if ((event == CELL_AVC2_EVENT_LOAD_FAILED ||
+							event == CELL_AVC2_EVENT_UNLOAD_SUCCEEDED ||
+							event == CELL_AVC2_EVENT_UNLOAD_FAILED) &&
+						error_code < 2)
+					{
+						sysutil_unregister_cb_with_id<avc2_cb_handle_t>();
+
+						std::lock_guard lock(this->mutex_cb);
+						this->avc2_cb = vm::null;
+						this->avc2_cb_arg = vm::null;
```

Comment:
Shouldn't it empty the args before calling the callback? (in the previous lock scope)

Reply:
It's accurate to the dispatcher of avc2(except for the addition of the mutex which I added for sanity sake)

Reply:
I dont think that it resets these if you only unload the video chat.

Reply:
I mean the entire unregister and reset of the callbacks

Reply:
There is no video callback and you can keep using the voice chat after unloading the video chat

Reply:
I changed it to return 2 for the error when unloading video which is probably why that check is there in the dispatcher.

Reply:
I kinda still doubt that it's the correct solution, but I guess it's good enough for now